### PR TITLE
Add missing virtual destructors

### PR DESCRIPTION
**Pull request by ivanpauno**
Extracted from https://github.com/ros2/geometry2/pull/270/, to see if we can merge this more trivial PR before the release.

### CODE REVIEW

**Comment by ivanpauno** [comment]
@jacobperron I think that relying on the PR checker for this is ok.
I don't have more time today for running CI or merging, so feel free to do so if you think this is ready.

**Review by jacobperron — APPROVED** [review]

File: tf2/include/tf2/buffer_core_interface.h [hunk 1]
```diff
@@ -50,6 +50,10 @@ namespace tf2
 class BufferCoreInterface
 {
 public:
+  TF2_PUBLIC
+  virtual
+  ~BufferCoreInterface() {}
```
Comment (wjwwood) [inline tf2/include/tf2/buffer_core_interface.h]: Is it more idiomatic to use `= default;` in this case? I honestly don't know, I'm just curious.

**Review by wjwwood — APPROVED** [review]
lgtm, with one small, non-blocking comment.

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10986/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6322/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8945/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10894/)

File: tf2/include/tf2/buffer_core_interface.h (hunk 1 above)
Comment (ivanpauno) [reply tf2/include/tf2/buffer_core_interface.h]: `= default` is definitely more idiomatic, I forgot about using it.

The only difference I know is that a class with a defaulted destructor is trivially destructible, and one with an empty destructor is not.
In this case it doesn't apply, as classes with virtual destructors are never trivially destructible, i.e. I think that a empty virtual destructor is exactly equivalent to the defaulted one. CPP is too complex so there might be another difference :joy:.

Comment (ivanpauno) [reply tf2/include/tf2/buffer_core_interface.h]: fixed in https://github.com/ros2/geometry2/pull/272/commits/143226d1fafc6e3a950cff010f2846849387fc92.

**Comment by ivanpauno** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10993/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6329/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8953/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10915/)

**Review by jacobperron — APPROVED** [review]

**Pull request merged**